src/blocks/modal.css

```diff
@@ -1,178 +1,208 @@
 .modal {
+  display: flex;
   position: fixed;
   top: 0;
   left: 0;
   bottom: 0;
   right: 0;
   background-color: rgba(0, 0, 0, 0.5);
-  z-index: 99;
-  align-items: center;
-  justify-content: center;
-  display: flex;
-  visibility: hidden;
+  z-index: 999;
+  padding: 0 19px;
+  visibility: none;
+  pointer-events: none;
   opacity: 0;
-  transition: visibility 0s linear 0.5s, opacity 0.5s linear;
+  transition: visibility 0s 0.6s, opacity 0.6s linear;
+  transition: visibility 0s 0.6s, opacity 0.6s linear;
 }
 
 .modal_opened {
-  display: flex;
   visibility: visible;
+  pointer-events: all;
   opacity: 1;
   transition-delay: 0s;
+  transition: opacity 0.5s linear; /* add transition for opacity */
 }
 
 .modal__container {
   background-color: #fff;
-  color: #000;
-  padding: 36px 36px 48px 36px;
-  width: 358px;
-  height: 258px;
+  margin: auto;
+  padding: 36px;
   border-radius: 10px;
   position: relative;
+  max-width: 430px;
+  display: flex;
+  flex-direction: column;
+}
+
+.modal__container_type_confirmation {
+  width: 358px;
+  height: 181px;
+}
+
+.modal__container_type_image {
+  background-color: transparent;
+  display: flex;
+  max-width: 75vw;
+  max-height: 75vh;
+  width: auto;
+  height: auto;
+  padding: 0;
+}
+
+.modal__heading {
+  font-family: "Inter", Arial, sans-serif;
+  font-style: normal;
+  font-weight: 900;
+  font-size: 24px;
+  line-height: 29px;
+  color: #000;
+  margin: 0 0 24px;
 }
 
 .modal__close {
-  border: none;
-  position: absolute;
-  bottom: calc(100% + 5px);
-  left: calc(100% + 5px);
   width: 40px;
   height: 40px;
   background-image: url("../images/close.icon.svg");
-  background-color: transparent;
   background-repeat: no-repeat;
   background-size: contain;
-  transition: opacity 0.1s linear;
+  border: none;
+  background-color: transparent;
+  cursor: pointer;
+  position: absolute;
+  bottom: calc(100% + 5px);
+  left: calc(100% + 5px);
 }
 
 .modal__close:hover {
-  opacity: 60%;
-  cursor: pointer;
+  opacity: 0.5;
 }
 
-.modal__heading {
-  margin: 0;
-  padding-bottom: 44px;
+.modal__form {
+  max-width: 358px;
```

Comment:
Needs correction: the avatar form width should not change when you enter something to the input

To fix that use `width` instead of `max-width`